Add Essent integration documentation

jaapp opened the pull request:
Silver-tier integration for monitoring dynamic energy prices in the Netherlands.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

Add documentation for new Essent integration for monitoring dynamic energy prices in the Netherlands.
Full disclosure: I work at Essent, but this integration is my personal contribution and is not officially provided by Essent N.V.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/157010
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/8511
- This PR fixes or closes issue: fixes N/A

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

jaapp committed:
Add Essent integration documentation

Silver-tier integration for monitoring dynamic energy prices in the Netherlands.

jaapp committed:
Merge branch 'next' into add-essent-integration-fixed

jaapp committed:
Merge branch 'next' into add-essent-integration-fixed

jaapp committed:
Essent docs: streamline troubleshooting

jaapp committed:
Document Essent price breakdown sensors

jaapp committed:
Document Essent next price sensors default state

jaapp committed:
Merge branch 'next' into add-essent-integration-fixed

joostlek commented on source/_integrations/essent.markdown:
```diff
@@ -0,0 +1,109 @@
+
+Prices are fetched from Essent's public API endpoint:
+
+`https://www.essent.nl/api/public/tariffmanagement/dynamic-prices/v1/`
```
We don't really have to explain where we get the data from

joostlek commented on source/_integrations/essent.markdown:
```diff
@@ -0,0 +1,109 @@
+
+- **Contract type**: Designed for Essent dynamic pricing contracts
+- **Data availability**: Current day and next day only (when available)
+- **No historical data**: Past pricing data is not stored or available - use HA recorder to capture these.
```
I don't think these are limitations with the integration, as in it feels kinda logical

jaapp committed:
Cleanup, sensor table format, add example

jaapp committed:
Fix API polling interval and remove redundant note

- Correct API polling from 'once per hour' to 'every 12 hours' (matches UPDATE_INTERVAL in code)
- Remove redundant note about dynamic pricing contract requirement (already stated in intro)

jaapp commented on source/_integrations/essent.markdown:
```diff
@@ -0,0 +1,119 @@
+alias: Charge EV at lowest price
+triggers:
+  - trigger: template
+    value_template: >-
+      {{ states('sensor.essent_current_electricity_price') ==
+      states('sensor.essent_lowest_electricity_price_today') }}
```
Invalid. Length is only 64 characters

jaapp committed:
addressed copilot review

c0ffeeca7 commented on source/_integrations/essent.markdown:
```diff
@@ -0,0 +1,116 @@
+
+The following sensors are created for electricity prices:
+
+- **Current electricity price** - Current hourly electricity price per kWh including VAT
+- **Next electricity price** - Next hour's electricity price per kWh (disabled by default)
+- **Average electricity price today** - Average of all hourly electricity prices for today
+- **Lowest electricity price today** - Minimum electricity price for today (disabled by default)
+- **Highest electricity price today** - Maximum electricity price for today (disabled by default)
+- **Current electricity price excl. VAT** - Current hourly price excluding VAT (disabled by default)
+- **Current electricity VAT** - VAT component of the current price (disabled by default)
+- **Current electricity market price** - Spot market component of the current price (disabled by default)
+- **Current electricity purchasing fee** - Supplier purchasing fee component (disabled by default)
+- **Current electricity tax** - Tax component of the current price (disabled by default)
```
```suggestion
- **Current electricity price** - Current hourly electricity price per kWh including VAT
- **Next electricity price** - Next hour's electricity price per kWh (disabled by default)
- **Average electricity price today**: Average of all hourly electricity prices for today
- **Lowest electricity price today**: Minimum electricity price for today (disabled by default)
- **Highest electricity price today**: Maximum electricity price for today (disabled by default)
- **Current electricity price excl. VAT**: Current hourly price excluding VAT (disabled by default)
- **Current electricity VAT** - VAT component of the current price (disabled by default)
- **Current electricity market price**: Spot market component of the current price (disabled by default)
- **Current electricity purchasing fee**: Supplier purchasing fee component (disabled by default)
- **Current electricity tax**: Tax component of the current price (disabled by default)
```

c0ffeeca7 commented on source/_integrations/essent.markdown:
```diff
@@ -0,0 +1,116 @@
+
+The following sensors are created for gas prices:
+
+- **Current gas price** - Current daily gas price per m³ including VAT
+- **Next gas price** - Next day's gas price per m³ (disabled by default)
+- **Current gas price excl. VAT** - Current daily price excluding VAT (disabled by default)
+- **Current gas VAT** - VAT component of the current price (disabled by default)
+- **Current gas market price** - Spot market component of the current price (disabled by default)
+- **Current gas purchasing fee** - Supplier purchasing fee component (disabled by default)
+- **Current gas tax** - Tax component of the current price (disabled by default)
```
```suggestion
- **Current gas price**: Current daily gas price per m³ including VAT
- **Next gas price**: Next day's gas price per m³ (disabled by default)
- **Current gas price excl. VAT**: Current daily price excluding VAT (disabled by default)
- **Current gas VAT**: VAT component of the current price (disabled by default)
- **Current gas market price**: Spot market component of the current price (disabled by default)
- **Current gas purchasing fee**: Supplier purchasing fee component (disabled by default)
- **Current gas tax**: Tax component of the current price (disabled by default)
```

c0ffeeca7 commented on source/_integrations/essent.markdown:
```diff
@@ -0,0 +1,116 @@
+
+If your sensors are showing unavailable or unknown states, check the following:
+
+1. **Network connectivity**: Verify that Home Assistant can reach `essent.nl`
+   - Check your network and firewall settings
+   - Verify internet connectivity
```
```suggestion
1. Network connectivity: Verify that Home Assistant can reach `essent.nl`.
   - Check your network and firewall settings.
   - Verify internet connectivity.
```
We use bold to mark UI strings. If possible, we try to avoid using it for emphasis.

frenck reviewed (approved):
Thanks, @jaapp 👍 

../Frenck  

<p>
  <a href="https://www.linkedin.com/in/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/linkedin.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://youtube.com/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/youtube.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://bsky.app/profile/frenck.social"><img src="https://github.com/frenck/frenck/raw/main/images/bluesky.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://fosstodon.org/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/mastodon.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.instagram.com/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/instagram.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.threads.net/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/threads.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.facebook.com/frenck.dev/"><img src="https://github.com/frenck/frenck/raw/main/images/facebook.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://x.com/frenck"><img src="https://github.com/frenck/frenck/raw/main/images/x.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.tiktok.com/@frenck.nl"><img src="https://github.com/frenck/frenck/raw/main/images/tiktok.svg" width="18" height="18"></a>
</p>

Blogging my personal ramblings at <a href="https://frenck.dev">frenck.dev</a>

c0ffeeca7 commented on source/_integrations/essent.markdown:
```diff
@@ -0,0 +1,116 @@
+   - Check your network and firewall settings
+   - Verify internet connectivity
+
+2. **API service status**: Essent's API may be temporarily unavailable
+   - Check {% my logs title="Settings → System → Logs" %} for error messages
+   - Open `https://www.essent.nl/api/public/tariffmanagement/dynamic-prices/v1/` in a browser or via `curl`; a JSON response confirms the service is reachable, while HTTP errors mean the API is down or blocked by your network
+   - Wait and check if data returns within an hour
```
```suggestion
2. **API service status**: Essent's API may be temporarily unavailable.
   - Check {% my logs title="Settings → System → Logs" %} for error messages.
   - Open `https://www.essent.nl/api/public/tariffmanagement/dynamic-prices/v1/` in a browser or via `curl`; a JSON response confirms the service is reachable, while HTTP errors mean the API is down or blocked by your network.
   - Wait and check if the data returns within an hour.
```
